feat: allow running single test (#3619)
scalameta/metals: docs/editors/overview.md

```

> 

Reviewer: Could you run prettier on the file?

Browse packages, classes and methods in the workspace and library dependencies
using the Metals sidebar. This feature is only implemented in VS Code.

## Test Explorer

Test Explorer is a feature that allows editors to display tests as a separate tree representation of tests. 
Although it was implemented in order to use Visual Studio Code's [Testing API](https://code.visualstudio.com/api/extension-guides/testing). Test Explorer API is editor agnostic and can be used by other editors than just VS Code.
![test-explorer](https://i.imgur.com/Z3VtS0O.gif)

Work on Test Explorer is still on progress and the feature has some known limitations:

> - Test Explorer is able to discover single test cases only for JUnit4 test classes. Support for other test framework is being worked on.

Reviewer 2: ```suggestion
- Test Explorer is able to discover single test cases only for JUnit4 test classes. Support for other test frameworks is being worked on.
```

Browse packages, classes and methods in the workspace and library dependencies
using the Metals sidebar. This feature is only implemented in VS Code.

## Test Explorer

> Test Explorer is a feature that allows editors to display tests in workspace in a different way than code lenses. Altough it was implemented in order to use Visual Studio Code's [Testing API](https://code.visualstudio.com/api/extension-guides/testing). Test Explorer API is editor agnostic and can be used by other editors than just VS Code.

Reviewer: ```suggestion
Test Explorer is a feature that allows editors to display tests as a separate tree representation of tests. Although it was implemented in order to use Visual Studio Code's [Testing API](https://code.visualstudio.com/api/extension-guides/testing). Test Explorer API is editor agnostic and can be used by other editors than just VS Code.
```

![test-explorer](https://i.imgur.com/Z3VtS0O.gif)

> Work on Test Explorer is still on progress and feature has known limitations:

Reviewer: ```suggestion
Work on Test Explorer is still on progress and the feature has some known limitations:
```

> - Test Explorer is able to discover single test cases only for JUnit4 test classes. Support for other test framework is in progress. However, running single tests is available only when you use Bloop as build server. Otherwise running single test will result in running whole suite.

Reviewer: ```suggestion
- Test Explorer is able to discover single test cases only for JUnit4 test classes. Support for other test framework is being worked on. 
```
I wouldn't even mention it if we would add it to sbt soon.

- detecting suites in cross scala-version projects is inconsistent, see [this issue](https://github.com/scalameta/metals/issues/3503).

> - there is no support for JS and Native platforms

Reviewer: Let's add a feature request for it so that people can follow the issue.

You can find more
information about Test Explorer under the [VS Code](vscode.md#test-explorer) specific section.

> ### Test Discovery

Reviewer: This section is specific to the implementation, it's not interesting for the users. I would add it to https://scalameta.org/metals/docs/integrations/new-editor or a separate section altogether
